hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/RpcConnection.java

```diff
@@ -89,7 +90,18 @@ protected RpcConnection(Configuration conf, HashedWheelTimer timeoutTimer, Conne
       MetricsConnection metrics) throws IOException {
     this.timeoutTimer = timeoutTimer;
     this.codec = codec;
-    this.compressor = compressor;
+    if (compressor != null) {
+      // Only enable compression for remote rpcs.
+      InetSocketAddress remoteAddr = Address.toSocketAddress(remoteId.getAddress());
```

Comment:
nit: this would be easier to read if you extract the `isLocal(ConnectionId)` logic into an external method. Then you can have a simpler conditional clause. Something like,

```
    this.compressor = isLocal(remoteId) ? null : compressor;
```

Reply:
will do.